core/client/app/components/gh-menu-toggle.js

```diff
@@ -0,0 +1,46 @@
+/*
+This cute little component has two jobs.
+
+On desktop, it toggles autoNav behaviour. It tracks
+that state via the maximise property, and uses the
+state to render the appropriate icon.
+
+On mobile, it renders a closing icon, and clicking it
+closes the mobile menu
+*/
+
+import Ember from 'ember';
+import mobileQuery from 'ghost/utils/mobile';
+
+export default Ember.Component.extend({
+    classNames: ['gh-menu-toggle'],
+
+    isMobile: false,
+    maximise: false,
+
+    iconClass: Ember.computed('maximise', 'isMobile', function () {
+        if (this.get('maximise') && !this.get('isMobile')) {
+            return 'icon-maximise';
+        } else {
+            return 'icon-minimise';
+        }
+    }),
+
+    init () {
+        var self = this;
+        this._super(...arguments);
+        this.set('isMobile', mobileQuery.matches);
+        mobileQuery.addListener(() =>
+            Ember.run.next(() => self.set('isMobile', mobileQuery.matches))
+        );
+    },
```

Comment:
Does this need to be in `init()`?  init is going private after 1.12 so if we can use the normal lifecycle hooks I think that'd be safer...  Something like:

```javascript
willInsertElement: function () {
    this.set('isMobile', mobileQuery.matches);

    this.set('mqlListener', Ember.run.bind(this, function (mql) {
        this.set('isMobile', mql.matches);
    }));

    mobileQuery.addListener(this.get('mqlListener'));
},

willDestroyElement: function () {
    mobileQuery.removeListener(this.get('mqlListener'));
}
```